tests/cpp/test_memory.cpp

```diff
@@ -2504,6 +2506,74 @@ TEST_P(LdMatrixTest, Regular) {
   testValidate(&fusion, cg_outputs, {t0}, __LINE__, __FILE__);
 }
 
+class StMatrixTest : public NVFuserTest {};
+
+TEST_F(StMatrixTest, Regular) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto operand = MmaOperand::B;
+
+  int sizeM = 8;
+  int sizeN = 8;
+
+  auto tv0 = makeContigConcreteTensor({sizeM, sizeN}, DataType::Half);
+  fusion.addInput(tv0);
+  // tv0 (global) -> tv1 (shared)
+  auto tv1 = set(tv0);
+  tv1->setMemoryType(MemoryType::Shared);
+  auto tv2 = set(tv1);
+  // tv1 (shared) -> tv2 (registers)
+  // tv2 (registers) -> tv3 (shared)
+  auto tv3 = set(tv2);
+  tv3->definition()->as<LoadStoreOp>()->setOpType(LoadStoreOpType::StMatrix);
+  tv3->setMemoryType(MemoryType::Shared);
+  // tv3 (shared) -> tv4(global)
+  auto tv4 = set(tv3);
+  fusion.addOutput(tv4);
+
+  tv2->applyMmaSwizzle(operand);
+  tv2->setAllocationDomain(tv2->getLoopDomain(), true);
+
+  // Scheduler tv2 as operand B in scheduleLdMatrix
+
+  //[ni8, koi4, koo1, no1, ki2] -> [koo1, no1, ni8, koi4, ki2]
+  tv2->reorder({{-2, -4}, {-3, -5}});
+
+  //  [koo1, no1, ni8, koi4, ki2] ->  [koo1, no1, ni8, 8]
+  tv2->merge(-2);
+  //  [1, ni8, 8]
+  tv2->merge(-4);
+  //  [8, 8]
+  tv2->merge(-3);
+
+  int64_t num_tidx_with_addr =
+      tv2->axis(-2)->extent()->evaluate().as<int64_t>();
+  if (num_tidx_with_addr < 32) {
+    int64_t factor = 32 / num_tidx_with_addr;
+    // [8, 8] -> [8, 4, 2]
+    tv2->split(-1, factor, false);
+    // [8, 4, 2] -> [4, 8, 2]
+    tv2->reorder({{-2, -3}, {-3, -2}});
+    // [32, 4]
+    tv2->merge(-3);
+  }
+  tv2->axis(-2)->parallelize(ParallelType::TIDx);
+  // We do this only because codegen depends on this
+  tv2->axis(-1)->parallelize(ParallelType::Vectorize);
```

Comment:
Is this part needed? Because we are not using ldmatrix to load, so we can just use a very simple loop domain to load, instead of scheduling the same way as ldmatrix? Why not just make the loop domain and allocation domain the same?